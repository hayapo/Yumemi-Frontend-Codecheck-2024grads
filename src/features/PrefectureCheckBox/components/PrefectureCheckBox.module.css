.prefecture_checkbox_container {
  display: grid;
  grid-row-gap: 1rem;
}

/* 都道府県選択チェックボックスのスタイル */
.prefecture_checkboxes {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));
  gap: 0.1rem;
  flex-wrap: wrap;
  align-items: center;
}
