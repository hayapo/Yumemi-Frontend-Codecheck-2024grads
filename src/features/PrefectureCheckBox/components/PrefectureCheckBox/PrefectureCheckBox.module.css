/* 都道府県選択チェックボックスのスタイル */
.prefecture_checkboxes {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));
  gap: 0.1rem;
  flex-wrap: wrap;
  margin: 0 1rem;
  align-items: center;
}
