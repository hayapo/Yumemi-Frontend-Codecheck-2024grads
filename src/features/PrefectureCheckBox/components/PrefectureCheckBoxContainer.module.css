.prefecture_checkbox_container {
  display: grid;
  grid-row-gap: 1rem;
}
