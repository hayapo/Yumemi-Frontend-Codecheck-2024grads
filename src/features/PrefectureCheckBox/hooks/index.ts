export { usePrefectureCheck } from "./usePrefectureCheck"
export { usePrefectures } from "./usePrefectures"
