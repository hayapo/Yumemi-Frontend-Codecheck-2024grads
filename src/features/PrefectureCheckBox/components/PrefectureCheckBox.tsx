import React from "react"
import { CheckBox } from "@/components"
import style from "./PrefectureCheckBox.module.css"
import { usePrefectureCheck } from "../lib/hooks"
import { usePrefectureWithCheckedContext } from "@/contexts"

/**
 * 都道府県のチェックボックスコンポーネント \
 * usePrefectures()で取得した都道府県一覧を使用する
 * @returns {React.FC<{}>}
 */
const PrefectureCheckBox: React.FC = () => {
  const toggleCheck = usePrefectureCheck()
  const { prefectureWithChecked } = usePrefectureWithCheckedContext()
  return (
    <>
      <h2 className={style.section_title}>都道府県</h2>
      <div className={style.prefecture_checkbox}>
        {prefectureWithChecked.map(({ prefCode, prefName }) => (
          <CheckBox label={prefName} id={`pref_${prefCode}`} key={prefCode} onChange={() => toggleCheck(prefCode)} />
        ))}
      </div>
    </>
  )
}

export default React.memo(function PrefectureCheckBoxSection() {
  return (
    <section id="prefecture_checkbox_section">
      <PrefectureCheckBox />
    </section>
  )
})
