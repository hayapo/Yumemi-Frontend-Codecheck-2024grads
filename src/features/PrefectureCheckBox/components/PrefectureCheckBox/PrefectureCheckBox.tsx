import React from "react"
import { isMobile } from "react-device-detect"
import { CheckBox } from "@/components"
import { PrefectureCheckBoxAccordion } from "../PrefectureCheckBoxAccordion"
import { usePrefectureCheck } from "../../lib/hooks"
import { usePrefectureWithCheckedContext } from "@/contexts"
import { SectionTitle } from "@/components/SectionTitle/SectionTitle"
import style from "./PrefectureCheckBox.module.css"

/**
 * 都道府県のチェックボックスコンポーネント \
 * usePrefectures()で取得した都道府県一覧を使用する
 * @returns {React.FC<{}>}
 */
const PrefectureCheckBox: React.FC = () => {
  const toggleCheck = usePrefectureCheck()
  const { prefectureWithChecked } = usePrefectureWithCheckedContext()

  return (
    <>
      <div className={style.prefecture_checkbox_container}>
        <SectionTitle title="都道府県" />
        {isMobile ? (
          <PrefectureCheckBoxAccordion prefectureWithChecked={prefectureWithChecked} toggleCheck={toggleCheck} />
        ) : (
          <div className={style.prefecture_checkboxes}>
            {prefectureWithChecked.map(({ prefCode, prefName, checked }) => (
              <CheckBox
                label={prefName}
                checked={checked}
                id={`pref_${prefCode}`}
                key={prefCode}
                onChange={() => toggleCheck(prefCode)}
              />
            ))}
          </div>
        )}
      </div>
    </>
  )
}

export default React.memo(function PrefectureCheckBoxSection() {
  return (
    <section id="prefecture_checkbox_section">
      <PrefectureCheckBox />
    </section>
  )
})
