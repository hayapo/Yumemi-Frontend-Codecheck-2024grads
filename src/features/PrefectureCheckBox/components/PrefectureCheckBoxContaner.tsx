import { Suspense, lazy } from "react"
import { isMobile } from "react-device-detect"
import { usePrefectureCheck } from "../hooks"
import { usePrefecturesWithCheckedContext } from "@/contexts"
import { Fallback, SectionTitle } from "@/components"

import style from "./PrefectureCheckBoxContainer.module.css"

// CheckBoxコンポーネントをDynamic Import
const PrefectureCheckBox = lazy(() => import("./PrefectureCheckBox"))
const PrefectureCheckBoxAccordion = lazy(() => import("./PrefectureCheckBoxAccordion"))

/**
 * 都道府県チェックボックスコンポーネントを集めたコンテナー \
 * `isMobile`でスマホかデスクかの判定をしてコンポーネントを出し分けている
 */
export const PrefectureCheckBoxContainer = () => {
  const toggleCheck = usePrefectureCheck()
  const { prefecturesWithChecked } = usePrefecturesWithCheckedContext()

  return (
    <div className={style.prefecture_checkbox_container}>
      <SectionTitle title="都道府県" />
      <Suspense fallback={<Fallback />}>
        {isMobile ? (
          <PrefectureCheckBoxAccordion prefecturesWithChecked={prefecturesWithChecked} toggleCheck={toggleCheck} />
        ) : (
          <PrefectureCheckBox prefecturesWithChecked={prefecturesWithChecked} toggleCheck={toggleCheck} />
        )}
      </Suspense>
    </div>
  )
}
