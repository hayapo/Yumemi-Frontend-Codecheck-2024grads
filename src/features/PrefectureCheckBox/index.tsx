import { memo } from "react"
import { PrefectureCheckBoxContainer } from "./components/PrefectureCheckBoxContaner"

export default memo(function PrefectureCheckBoxSection() {
  return (
    <>
      <section id="prefecture_checkbox_section">
        <PrefectureCheckBoxContainer />
      </section>
    </>
  )
})
