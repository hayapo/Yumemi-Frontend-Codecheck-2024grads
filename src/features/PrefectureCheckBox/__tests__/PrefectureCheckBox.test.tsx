// @vitest-environment jsdom
import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import { CheckBox } from "@/components"
import { mockPrefecturesData } from "@/mocks/mockPrefecturesData"
import { default as PrefectureCheckBoxSection } from "../../PrefectureCheckBox"
import { PrefecturesWithCheckedProvider } from "@/contexts"

test("should render <CheckBox ./> correctly", () => {
  mockPrefecturesData.map(({ prefCode, prefName, checked }) => {
    const { container } = render(<CheckBox key={prefCode} label={prefName} checked={checked} id={`pref_${prefCode}`} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})

test("label is render correctly", () => {
  mockPrefecturesData.map(({ prefCode, prefName, checked }) => {
    render(<CheckBox key={prefCode} label={prefName} checked={checked} id={`pref_${prefCode}`} />)
    expect(screen.getByLabelText(prefName)).toBeInTheDocument()
  })
})

test("should render <PrefectureCheckBox /> correctly", () => {
  const { container } = render(
    <PrefecturesWithCheckedProvider>
      <PrefectureCheckBoxSection />
    </PrefecturesWithCheckedProvider>
  )
  expect(container.firstChild).toMatchSnapshot()
})

test("title is visible", () => {
  render(
    <PrefecturesWithCheckedProvider>
      <PrefectureCheckBoxSection />
    </PrefecturesWithCheckedProvider>
  )
  expect(screen.getByText("都道府県")).toBeInTheDocument()
})
