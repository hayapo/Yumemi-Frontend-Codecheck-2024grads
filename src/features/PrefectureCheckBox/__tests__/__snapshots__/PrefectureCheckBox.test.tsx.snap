// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correctly 1`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_1"
    type="checkbox"
  />
  <span>
    北海道
  </span>
</label>
`;

exports[`should render correctly 2`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_2"
    type="checkbox"
  />
  <span>
    青森県
  </span>
</label>
`;

exports[`should render correctly 3`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_3"
    type="checkbox"
  />
  <span>
    岩手県
  </span>
</label>
`;

exports[`should render correctly 4`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_4"
    type="checkbox"
  />
  <span>
    宮城県
  </span>
</label>
`;

exports[`should render correctly 5`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_5"
    type="checkbox"
  />
  <span>
    秋田県
  </span>
</label>
`;

exports[`should render correctly 6`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_6"
    type="checkbox"
  />
  <span>
    山形県
  </span>
</label>
`;

exports[`should render correctly 7`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_7"
    type="checkbox"
  />
  <span>
    福島県
  </span>
</label>
`;

exports[`should render correctly 8`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_8"
    type="checkbox"
  />
  <span>
    茨城県
  </span>
</label>
`;

exports[`should render correctly 9`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_9"
    type="checkbox"
  />
  <span>
    栃木県
  </span>
</label>
`;

exports[`should render correctly 10`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_10"
    type="checkbox"
  />
  <span>
    群馬県
  </span>
</label>
`;

exports[`should render correctly 11`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_11"
    type="checkbox"
  />
  <span>
    埼玉県
  </span>
</label>
`;

exports[`should render correctly 12`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_12"
    type="checkbox"
  />
  <span>
    千葉県
  </span>
</label>
`;

exports[`should render correctly 13`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_13"
    type="checkbox"
  />
  <span>
    東京都
  </span>
</label>
`;

exports[`should render correctly 14`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_14"
    type="checkbox"
  />
  <span>
    神奈川県
  </span>
</label>
`;

exports[`should render correctly 15`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_15"
    type="checkbox"
  />
  <span>
    新潟県
  </span>
</label>
`;

exports[`should render correctly 16`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_16"
    type="checkbox"
  />
  <span>
    富山県
  </span>
</label>
`;

exports[`should render correctly 17`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_17"
    type="checkbox"
  />
  <span>
    石川県
  </span>
</label>
`;

exports[`should render correctly 18`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_18"
    type="checkbox"
  />
  <span>
    福井県
  </span>
</label>
`;

exports[`should render correctly 19`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_19"
    type="checkbox"
  />
  <span>
    山梨県
  </span>
</label>
`;

exports[`should render correctly 20`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_20"
    type="checkbox"
  />
  <span>
    長野県
  </span>
</label>
`;

exports[`should render correctly 21`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_21"
    type="checkbox"
  />
  <span>
    岐阜県
  </span>
</label>
`;

exports[`should render correctly 22`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_22"
    type="checkbox"
  />
  <span>
    静岡県
  </span>
</label>
`;

exports[`should render correctly 23`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_23"
    type="checkbox"
  />
  <span>
    愛知県
  </span>
</label>
`;

exports[`should render correctly 24`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_24"
    type="checkbox"
  />
  <span>
    三重県
  </span>
</label>
`;

exports[`should render correctly 25`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_25"
    type="checkbox"
  />
  <span>
    滋賀県
  </span>
</label>
`;

exports[`should render correctly 26`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_26"
    type="checkbox"
  />
  <span>
    京都府
  </span>
</label>
`;

exports[`should render correctly 27`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_27"
    type="checkbox"
  />
  <span>
    大阪府
  </span>
</label>
`;

exports[`should render correctly 28`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_28"
    type="checkbox"
  />
  <span>
    兵庫県
  </span>
</label>
`;

exports[`should render correctly 29`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_29"
    type="checkbox"
  />
  <span>
    奈良県
  </span>
</label>
`;

exports[`should render correctly 30`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_30"
    type="checkbox"
  />
  <span>
    和歌山県
  </span>
</label>
`;

exports[`should render correctly 31`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_31"
    type="checkbox"
  />
  <span>
    鳥取県
  </span>
</label>
`;

exports[`should render correctly 32`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_32"
    type="checkbox"
  />
  <span>
    島根県
  </span>
</label>
`;

exports[`should render correctly 33`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_33"
    type="checkbox"
  />
  <span>
    岡山県
  </span>
</label>
`;

exports[`should render correctly 34`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_34"
    type="checkbox"
  />
  <span>
    広島県
  </span>
</label>
`;

exports[`should render correctly 35`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_35"
    type="checkbox"
  />
  <span>
    山口県
  </span>
</label>
`;

exports[`should render correctly 36`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_36"
    type="checkbox"
  />
  <span>
    徳島県
  </span>
</label>
`;

exports[`should render correctly 37`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_37"
    type="checkbox"
  />
  <span>
    香川県
  </span>
</label>
`;

exports[`should render correctly 38`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_38"
    type="checkbox"
  />
  <span>
    愛媛県
  </span>
</label>
`;

exports[`should render correctly 39`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_39"
    type="checkbox"
  />
  <span>
    高知県
  </span>
</label>
`;

exports[`should render correctly 40`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_40"
    type="checkbox"
  />
  <span>
    福岡県
  </span>
</label>
`;

exports[`should render correctly 41`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_41"
    type="checkbox"
  />
  <span>
    佐賀県
  </span>
</label>
`;

exports[`should render correctly 42`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_42"
    type="checkbox"
  />
  <span>
    長崎県
  </span>
</label>
`;

exports[`should render correctly 43`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_43"
    type="checkbox"
  />
  <span>
    熊本県
  </span>
</label>
`;

exports[`should render correctly 44`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_44"
    type="checkbox"
  />
  <span>
    大分県
  </span>
</label>
`;

exports[`should render correctly 45`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_45"
    type="checkbox"
  />
  <span>
    宮崎県
  </span>
</label>
`;

exports[`should render correctly 46`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_46"
    type="checkbox"
  />
  <span>
    鹿児島県
  </span>
</label>
`;

exports[`should render correctly 47`] = `
<label
  class="_checkboxContainer_e15d39"
>
  <input
    id="pref_47"
    type="checkbox"
  />
  <span>
    沖縄県
  </span>
</label>
`;

exports[`should render correctly 48`] = `
<h2
  class="_section_title_99397d"
>
  都道府県
</h2>
`;
